src/next-collection-view.js

```diff
@@ -0,0 +1,684 @@
+  // viewComparator logic
+  getViewComparator() {
+    return this.viewComparator || this._viewComparator;
+  },
+
+  // Default internal view comparator that order the views by
+  // the order of the collection
+  _viewComparator(view) {
+    if (!this.collection) { return; }
+    return this.collection.indexOf(view.model);
+  },
+
+  // This method re-filters the children views and re-renders the results
+  filter() {
+    if (this._isDestroyed) { return this; }
+
+    if (!this.children.length) { return this; }
+
+    const filteredViews = this._filterChildren();
+
+    this._renderChildren(filteredViews);
+
+    return this;
+  },
+
+  _filterChildren() {
+    const viewFilter = this._getFilter();
+
+    if (!viewFilter) {
+      return this.children._views;
+    }
+
+    this.triggerMethod('before:filter', this);
+
+    const filteredViews = this.children.partition(_.bind(viewFilter, this));
+
+    this._detachChildren(filteredViews[1]);
+
+    this.triggerMethod('filter', this);
+
+    return filteredViews[0];
+  },
+
+  // This method returns a function for the viewFilter
+  _getFilter() {
+    const viewFilter = this.getViewFilter();
+
+    if (!viewFilter) { return false; }
+
+    if (_.isFunction(viewFilter)) {
+      return viewFilter;
+    }
+
+    // Support filter predicates `{ fooFlag: true }`
+    if (_.isObject(viewFilter)) {
+      const matcher = _.matches(viewFilter);
+      return function(view) {
+        return matcher(view.model && view.model.attributes);
+      };
+    }
+
+    // Filter by model attribute
+    if (_.isString(viewFilter)) {
+      return function(view) {
+        return view.model && view.model.get(viewFilter);
+      };
+    }
+
+    throw new MarionetteError({
+      name: 'InvalidViewFilterError',
+      message: '"viewFilter" must be a function, predicate object literal, a string indicating a model attribute, or falsy'
+    });
+  },
+
+  // Override this function to provide custom
+  // viewFilter logic
+  getViewFilter() {
+    return this.viewFilter;
+  },
+
+  // Sets the view's `viewFilter` and applies the filter if the view is ready.
+  // To prevent the render pass `{ preventRender: true } as the 2nd argument.
+  setFilter(filter, {preventRender} = {}) {
+    const filterChanged = this.viewFilter !== filter;
+    const shouldRender = filterChanged && !preventRender;
+
+    this.viewFilter = filter;
+
+    if (shouldRender) {
+      this.filter();
+    }
+
+    return this;
+  },
+
+  // Clears the `viewFilter` and follows the same rules for rendering as `setFilter`.
+  removeFilter(options) {
+    return this.setFilter(null, options);
+  },
+
+  _detachChildren(detachingViews) {
+    _.each(detachingViews, _.bind(this._detachChildView, this));
+  },
+
+  _detachChildView(view) {
+    const shouldTriggerDetach = !!view._isAttached;
+    if (shouldTriggerDetach) {
+      triggerMethodOn(view, 'before:detach', view);
+    }
+
+    this.detachHtml(view);
```

Comment:
This'll still detach this child view from this view without destroying the child.  the detach event is only concerned with if the view is in the DOM.. not if it is attached to some `el`.  I don't think we want to prevent child view management if the collectionview isn't in the DOM.